### PR TITLE
fix race condition with paper bins

### PR DESCRIPTION
**Pull request by amylizzle**
## About the PR 
Adds a check to make sure the item is on the ground before we try to eat it, in case someone already picked it up.

### CODE REVIEW

File: code/mob/living/critter/ai/flock/flocktasks.dm [hunk 1]
```diff
@@ -659,6 +659,10 @@ stare
 
 /datum/aiTask/succeedable/harvest/on_tick()
 	var/obj/item/harvest_target = holder.target
+	if(!harvest_target || !isturf(harvest_target.loc)) //sometimes there are race conditions, and we should deal with that
+		harvest_target = null
```
Comment (stonepillars) [inline code/mob/living/critter/ai/flock/flocktasks.dm:663]: Redundant assignment to null right before a return?

Comment (amylizzle) [reply code/mob/living/critter/ai/flock/flocktasks.dm:663]: not redundant, nulls out harvest_target to trigger failure() on the next evaluation and reset the task.

Comment (TobleroneSwordfish) [reply code/mob/living/critter/ai/flock/flocktasks.dm:663]: surely harvest_target is reset anyway due to being a local variable?

Comment (stonepillars) [reply code/mob/living/critter/ai/flock/flocktasks.dm:663]: You probably meant to null out `holder.target`, you mutated the value of `harvest_target`

Comment (amylizzle) [reply code/mob/living/critter/ai/flock/flocktasks.dm:663]: oops, that is exactly what I meant to do. willfix.

Comment (amylizzle) [reply code/mob/living/critter/ai/flock/flocktasks.dm:663]: oops, that is exactly what I meant to do. willfix.

**Pull request merged**